src/webserver/HttpBase.h

```diff
@@ -260,10 +281,16 @@ class HttpBase : public util::Taggable
                 workQueue_);
         }
 
-        auto ip = derived().ip();
-
-        if (!ip)
-            return;
+        // to avoid overwhelm work queue, the request limit check should be
+        // before posting to queue the web socket creation will be guarded via
+        // connection limit
+        if (!dosGuard_.request(ip.value()))
```

Comment:
don't post to work queue if dosguard fail

Reply:
If `ip` is nullopt for whatever reason (probably never?) `.value()` will crash the app entirely. I think Maybe we should leave that check with early return in here?

Reply:
IP valid check is done at line 243